x-pack/plugins/triggers_actions_ui/public/application/sections/rules_list/components/rules_list.tsx

```diff
@@ -661,8 +676,21 @@ export const RulesList = ({
     isSchedulingRules,
     isUnschedulingRules,
     isUpdatingRuleAPIKeys,
+    isCloningRule,
   ]);
 
+  const onCloneRule = async (clone: () => Promise<string>) => {
```

Comment:
you could consider moving everything cloning related to its own hook to improve readability of this component/separation of concerns